src/components/login.js

```diff
@@ -1,17 +1,84 @@
-import React from "react";
+import React, { useEffect, useState, useRef } from "react";
 import { Link } from "react-router-dom";
+import * as yup from "yup";
+import { useForm } from "react-hook-form";
+import { yupResolver } from "@hookform/resolvers/yup";
+import getAllUsers from "../common/get-all-users";
+import getUserByAccountNumber from "../common/get-user-act-no";
+import { useNavigate } from "react-router-dom";
+
 export const Login = () => {
+  const [option, setOption] = useState([]);
+  const [inputValue, setInputValue] = useState();
+  const navigate = useNavigate();
+  const selectRef = useRef(null);
+
+  const handleInputChange = (event) => {
+    setInputValue(event.target.value);
+  };
+
+  useEffect(() => {
+    const accountNames = getAllUsers();
+    const optionElem = accountNames.map((user, index) => (
+      <option key={index} value={user.accountNumber}>
+        {user.accountName} ({user.accountPin})
+      </option>
+    ));
+
+    setOption(optionElem);
+  }, []);
+  const schema = yup.object({
+    accountPin: yup.string().min(4).max(4).required("Account pin required"),
+  });
+
+  const {
+    register,
+    handleSubmit,
+    formState: { errors },
+  } = useForm({
+    resolver: yupResolver(schema),
+  });
+  const onSubmit = (data) => {
+    const accountNumber = selectRef.current.value;
+
+    console.log(accountNumber);
+    const existingUser = getUserByAccountNumber(accountNumber);
+
+    if (existingUser.accountPin !== data.accountPin) {
+      alert("Incorrect PIN");
+      return;
+    }
+    if (!existingUser) {
+      alert("Account not found");
+      return;
+    }
+    localStorage.setItem(
+      "MB_LOGGEDIN_USER_ACCOUNT_NUMBER",
+      JSON.stringify(accountNumber)
+    );
+    navigate("/transaction");
+  };
+
   return (
     <div className="container-fixed">
       <h1>Login</h1>
-      <form className="account-form">
+      <form className="account-form" onSubmit={handleSubmit(onSubmit)}>
         <div className="form-group">
-          <select className="form-control">
-            <option value="">Select Username</option>
+          <select ref={selectRef} className="form-control">
```

Comment:
No need to use ref here. You already have the list of your users in local storage, so it's already saved in a permanent state. All you need do is map and render the users in <option> element like you were doing in the useEffect function.